### PR TITLE
Managed instance restructured

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Restructured SQL Managed Instance to child-resource structure
- Updated ReadMe Generation Skript to better work with object references

Pipeline reference
[Sql: Managedinstances](https://github.com/Azure/ResourceModules/actions/workflows/ms.sql.managedinstances.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.Sql/managedInstances/databases/deploy.bicep [hunk 1]
```diff
@@ -152,46 +129,53 @@ resource managedInstanceDatabase 'Microsoft.Sql/managedInstances/databases@2020-
...
+    weeklyRetention: contains(backupLongTermRetentionPoliciesObj, 'weeklyRetention') ? backupLongTermRetentionPoliciesObj.weeklyRetention : 'P1M'
+    monthlyRetention: contains(backupLongTermRetentionPoliciesObj, 'monthlyRetention') ? backupLongTermRetentionPoliciesObj.monthlyRetention : 'P1Y'
+    yearlyRetention: contains(backupLongTermRetentionPoliciesObj, 'yearlyRetention') ? backupLongTermRetentionPoliciesObj.yearlyRetention : 'P5Y'
+  }
 }
 
-output managedInstanceDatabaseName string = managedInstanceDatabase.name
-output managedInstanceDatabaseResourceId string = managedInstanceDatabase.id
-output managedInstanceDatabaseResourceGroup string = resourceGroup().name
+output databaseName string = last(split(database.name, '/'))
```
Comment (rahalan) [inline arm/Microsoft.Sql/managedInstances/databases/deploy.bicep:179]: missing description

File: arm/Microsoft.Sql/managedInstances/databases/backupLongTermRetentionPolicies/readme.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+| Output Name | Type | Description |
+| :-- | :-- | :-- |
+| `backupLongTermRetentionPolicyName` | string | The name of the deployed database backup long-term retention policy |
+| `backupLongTermRetentionPolicyResourceGroup` | string | The resource group of the deployed database backup long-term retention policy |
+| `backupLongTermRetentionPolicyResourceId` | string | The resourceId of the deployed database backup long-term retention policy |
+
+## Template references
+
+- [Managedinstances/Databases/Backuplongtermretentionpolicies](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Sql/2021-02-01-preview/managedInstances/databases/backupLongTermRetentionPolicies)
```
[the remark this answers is not in the document]
Comment (MrMCake) [reply arm/Microsoft.Sql/managedInstances/databases/backupLongTermRetentionPolicies/readme.md]: I did - but to be fair: this is autogenerated and will be overwritten

File: arm/Microsoft.Sql/managedInstances/databases/backupShortTermRetentionPolicies/readme.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+| Output Name | Type | Description |
+| :-- | :-- | :-- |
+| `backupShortTermRetentionPolicyName` | string | The name of the deployed database backup short-term retention policy |
+| `backupShortTermRetentionPolicyResourceGroup` | string | The resource group of the deployed database backup short-term retention policy |
+| `backupShortTermRetentionPolicyResourceId` | string | The resourceId of the deployed database backup short-term retention policy |
+
+## Template references
+
+- [Managedinstances/Databases/Backupshorttermretentionpolicies](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Sql/2017-03-01-preview/managedInstances/databases/backupShortTermRetentionPolicies)
```
[the remark this answers is not in the document]
Comment (MrMCake) [reply arm/Microsoft.Sql/managedInstances/databases/backupShortTermRetentionPolicies/readme.md]: done, but same comment as above

File: arm/Microsoft.Sql/managedInstances/databases/deploy.bicep (hunk 1 above)
Comment (MrMCake) [reply arm/Microsoft.Sql/managedInstances/databases/deploy.bicep]: added

**Review by rahalan — APPROVED** [review]

**Pull request merged**